Push built image as docker v2s2

chmeliik opened the pull request:
CLOUDBLD-8136

By default, the podman build command uses the oci format. The podman
push command defaults to the format of the built image.

Explicitly specify v2s2, because some parts of the RH container pipeline
do not support oci yet.

Signed-off-by: Adam Cmiel <acmiel@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a Python type annotations added to new code
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates
- n/a New feature can be disabled from a configuration file

hjmodi commented:
should we document this somewhere that images are built as `oci` and pushed as `v2s2`?

chmeliik commented:
> should we document this somewhere that images are built as `oci` and pushed as `v2s2`?

Added to docstrings at least, I don't think there is a good place for this info in osbs.readthedocs yet

@rcerven I also updated the commands to follow the "options-first" format. It gets a bit awkward when adding options dynamically (`insert(-1, option)`), LMK what you think

rcerven commented:
> > should we document this somewhere that images are built as `oci` and pushed as `v2s2`?
> 
> Added to docstrings at least, I don't think there is a good place for this info in osbs.readthedocs yet
> 
> @rcerven I also updated the commands to follow the "options-first" format. It gets a bit awkward when adding options dynamically (`insert(-1, option)`), LMK what you think

well that podman pushes default oci image is podman implementation, for our users nothing changes, we always create v2s2

hjmodi commented on atomic_reactor/tasks/binary_container_build.py:
```diff
@@ -240,14 +241,16 @@ def build_container(
         :param insecure: disable --tls-verify?
         """
-        push_cmd = [*self._podman_remote_cmd, "push", str(dest_tag)]
+        push_cmd = [*self._podman_remote_cmd, "push", "--format=v2s2", str(dest_tag)]
```
```suggestion
        options = []
        if self._registries_authfile:
            options.append(f"--authfile={self._registries_authfile}")
        if insecure:
            options.append("--tls-verify=false")
         push_cmd = [*self._podman_remote_cmd, "push", *options, str(dest_tag)]
```

how about this?
